### ISSUE TITLE
[Developer] It would help if compile success or failure was denoted more visibly

### ISSUE THREAD

**Issue by mcdurdin** [keymanapp/keyman#2016]
Please make sure that at the end of compiling there is a "Succeeded. <#> Errors". Currently there is no notification at the end that things are ready to go.

via Doug Higby, Martin Hosken.

It's unclear here exactly what toolchain they were using and how they were invoking the compiler.

**Comment by mcdurdin** [comment]
@mhosken can you elaborate?

**Comment by mhosken** [comment]
When pressing compile keyboard in Keyman Developer one has to actually read the textual compiler messages to work out whether the keyboard compiled successfully or not. It would help users a lot if there were some non-textual cue that a keyboard build was sucessfull or otherwise, perhaps through the use of colour. For those who can't distinguish colour, there is still the textual explanation.

The reason for this is that users often go through a relatively rapid modify, compile, test loop at some stage in their keyboard and anything that can help smooth and speed up that loop would be appreciated. The speed of spotting the success of a compilation was identified intuitively as one current slow point in the loop.

Actually this issue makes no request as to how the user should quickly be able to tell that a keyboard compilation succeeded or failed, just that it is made much more obvious.

**Comment by mcdurdin** [comment]
Thanks @mhosken, that's very helpful. I'll reprioritise this as a v13 enhancement request 😉.

**Comment by DavidLRowe** [comment]
Even changing "successfully" to "SUCCESSFULLY" in the message would help (in my opinion).

**Comment by DavidLRowe** [comment]
Or even leaving the existing message as is, but putting "SUCCESS: " on the front, so the user can look at the left edge of the screen and quickly see the status. That is, change the current:
```
basic_kbdinguj.kmn: Compiling 'C:\Users\David\...
basic_kbdinguj.kmn: 'C:\Users\David\...
```
to:
```
basic_kbdinguj.kmn: Compiling 'C:\Users\David\...
basic_kbdinguj.kmn: SUCCESS: 'C:\Users\David\...
```
@mcdurdin Is it too late to get such a change into version 12?

**Comment by mcdurdin** [comment]
@DavidLRowe I'd prefer not to make a little change like that which may have other repercussions with the reporting -- I'd like to tidy it up properly and add colour. I'll see if I can do it this sprint but do have a bunch of other Keyman Developer bugs I need to tackle.

**Comment by mcdurdin** [comment]
@dhigby tagging you FYI as you were referenced in the original issue detail

**Comment by mcdurdin** [comment]
Resolved in 12.0.37
